### ISSUE TITLE
Do you have any idea why model didn't learn much?

### ISSUE THREAD

**Issue by jd0713** [wassname/rl-portfolio-management#2]
As you know, portfolio weights appear to static at test set.
Why didn't model learn so much?

I am assuming several reasons.

1. there are no ensemble in DDPG network
2. input doesn't include previous portfolio weight 
3. there are no online stochastic batch learning when applying network to test set 
4. different universe 
5. no PVM
6. Used ddpg instead of dpg

because author said that EIIE is there core idea in this paper, I think if I solve 1 and 2, there will be 
some improvement. 

Will you give some idea of yours? 

Thank you very much.

**Comment by wassname** [comment]
Are you using one of my notebooks or your own?

This happened for me when

1. The critic network wasn't large enough to model the reward, so I added more parameters
2. It learns from each actiion in sequence. But then the algorithm can just learn the current market and forget past markets. Lots of new algorithms try to fix this by either having multiple actors learning at once (I think the original paper did this), or using replay or prioritised replay memory. 
3. It failed to converge because SGD has the wrong parameters, in this case try RMSProp (or Adam but adam isn't as good for moving targets).

As for your list of guesses I think 2 and 5 couldn't be the cause since it makes it easier for it to give varied predictions. 3. shouldn't be it since it shouldn't be learning on the test set. But yeah I reckon it could be 1., 3., 4., or 6.

**Comment by jd0713** [comment]
I used your notebook, and data you uploaded. 
I mean that there are only slight difference between equal-weighted portfolio and learned policy. 
Also, there are almost no rewards gained while training. 
I think that portfolio weight, accumulated return, like these should be different if 'something' is learned, and result doesn't seemed to. 
I am wondering if there's way to model learn 'something'
Because I'm newbie at deep learning, my question might be vague.
Thanks.

**Comment by wassname** [comment]
I ran into similar problems, I found that the DDPG model was less likely to learn, but the VPG notebook was more stable. So my advice would be to try the VPG notebook and if that doesn't work maybe start of with with a simpler project. Deep reinforcement learning is pretty new and lots of libraries are broken and lots of algorithms are unstable. So maybe start with image classification projects if this is frustrating you?

If your determined to do this though, it's probably worth reading the papers belonging to the algorithms since they explain all the parameters and limitations.

This project didn't get good results for me (although another guy did a pytorch implementation and got decent results, but I'm not sure if he plans on publishing it). I'm wondering if there is a problem in my data or my environment since I tried many algorithms but kept the data and environment constant. Or maybe it's just difficult dataset.
